### PR TITLE
Remove bookmarksDao from BrowserTabViewModel

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/414730916066338/1200996756744658/f

### Description

Replaced the usage of bookmarksDao with bookmarksRepository in BrowserTabViewModel and updated BookmarksDataRepository with the needed functions.

### Steps to test this PR

Regression testing for the bookmark feature (add / edit / remove).

### CODE REVIEW

File: app/src/androidTest/java/com/duckduckgo/app/browser/BrowserTabViewModelTest.kt [hunk 1]
```diff
@@ -382,7 +380,6 @@ class BrowserTabViewModelTest {
             networkLeaderboardDao = mockNetworkLeaderboardDao,
             autoComplete = mockAutoCompleteApi,
             appSettingsPreferencesStore = mockSettingsStore,
-            bookmarksDao = mockBookmarksDao,
```
Comment (malmstein) [inline app/src/androidTest/java/com/duckduckgo/app/browser/BrowserTabViewModelTest.kt:385]: 💥

File: app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt [hunk 2]
```diff
@@ -1420,9 +1417,7 @@ class BrowserTabViewModel(
             if (url.isNotBlank()) {
                 faviconManager.persistCachedFavicon(tabId, url)
             }
-            val bookmarkEntity = BookmarkEntity(title = title, url = url, parentId = 0)
-            val id = bookmarksDao.insert(bookmarkEntity)
-            SavedSite.Bookmark(id, title, url, 0)
+            bookmarksRepository.insert(title, url)
```
Comment (malmstein) [inline app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt:1420]: 👍

**Review by malmstein — APPROVED** [review]
Thanks for this @anikiki! I've approved the PR pending that the tests are fixed. I ran them locally and seems to work, so might need to look at Bitrise

File: app/src/androidTest/java/com/duckduckgo/app/browser/BrowserTabViewModelTest.kt [hunk 3]
```diff
@@ -554,9 +551,10 @@ class BrowserTabViewModelTest {
     }
 
     @Test
-    fun whenBookmarkEditedThenDaoIsUpdated() = coroutineRule.runBlocking {
-        testee.onSavedSiteEdited(Bookmark(0, "A title", "www.example.com", 0))
-        verify(mockBookmarksDao).update(BookmarkEntity(title = "A title", url = "www.example.com", parentId = 0))
+    fun whenBookmarkEditedThenRepositoryIsUpdated() = coroutineRule.runBlocking {
+        val bookmark = Bookmark(0, "A title", "www.example.com", 0)
```
Comment (marcosholgado) [inline app/src/androidTest/java/com/duckduckgo/app/browser/BrowserTabViewModelTest.kt:555]: nit: for consistency (and also know what those 0 are) would be nice to add function parameters

```suggestion
        val bookmark = Bookmark(id = 0, title = "A title", url = "www.example.com", parentId = 0)
```

File: app/src/androidTest/java/com/duckduckgo/app/browser/BrowserTabViewModelTest.kt [hunk 4]
```diff
@@ -1505,11 +1502,12 @@ class BrowserTabViewModelTest {
 
     @Test
     fun whenNoSiteAndUserSelectsToAddBookmarkThenBookmarkIsNotAdded() = coroutineRule.runBlocking {
-        whenever(mockBookmarksDao.insert(any<BookmarkEntity>())).thenReturn(1)
+        val bookmark = Bookmark(0, "A title", "www.example.com", 0)
```
Comment (marcosholgado) [inline app/src/androidTest/java/com/duckduckgo/app/browser/BrowserTabViewModelTest.kt:1505]: nit: ditto

```suggestion
        val bookmark = Bookmark(id = 0, title = "A title", url = "www.example.com", parentId = 0)
```

**Review by malmstein — APPROVED** [review]
Thanks for this @anikiki!

**Pull request merged**